syntax = "proto3";

package api;

// Chat service
service AIService {
  rpc GetAIResponse (AIRequest) returns (AIResponse) {}
}

message AIRequest {
  string question = 1;
}

message AIResponse {
  string answer = 1;
}

// File processing service
service FileProcessingService {
  rpc ProcessFile (FileRequest) returns (FileResponse) {}
}

message FileRequest {
  bytes file = 1;
  string type = 2;
}

message FileResponse {
  string status = 1;
  string message = 2;
  repeated Test tests = 3;
}

message Test {
  string categories = 1;
  map<string, float> parameters = 2;
}